### PR TITLE
Dataflow: Revert support for flow-state based in-/out-barriers

### PR DESCRIPTION
**Pull request by aschackmull**
Resolves https://github.com/github/codeql/pull/8435#issuecomment-1087530306

### CODE REVIEW

**Comment by aschackmull** [comment]
Depends on https://github.com/github/codeql/pull/8702

File: cpp/ql/lib/change-notes/2022-03-14-flow-state-barriers.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
 ---
 category: feature
 ---
-* The data flow and taint tracking libraries have been extended with versions of `isBarrierIn`, `isBarrierOut`, and `isBarrierGuard`, respectively `isSanitizerIn`, `isSanitizerOut`, and `isSanitizerGuard`, that support flow states.
+* The data flow and taint tracking libraries have been extended with versions of `isBarrierGuard`, respectively `isSanitizerGuard`, that support flow states.
```
Comment (jketema) [inline cpp/ql/lib/change-notes/2022-03-14-flow-state-barriers.md:4]: I didn't keep track when the latest release was, but that might already have deleted this change note and the others like this..

File: cpp/ql/lib/change-notes/2022-04-08-flow-state-barriers.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+category: breaking
+---
+The recently added flow-state versions of `isBarrierIn`, `isBarrierOut`, `isSanitizerIn`, and `isSanitizerOut` in the data flow and taint tracking libraries have been removed.
```
Comment (jketema) [inline cpp/ql/lib/change-notes/2022-04-08-flow-state-barriers.md:4]:
```suggestion
* The recently added flow-state versions of `isBarrierIn`, `isBarrierOut`, `isSanitizerIn`, and `isSanitizerOut` in the data flow and taint tracking libraries have been removed.
```

File: csharp/ql/lib/change-notes/2022-04-08-flow-state-barriers.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+category: breaking
+---
+The recently added flow-state versions of `isBarrierIn`, `isBarrierOut`, `isSanitizerIn`, and `isSanitizerOut` in the data flow and taint tracking libraries have been removed.
```
Comment (jketema) [inline csharp/ql/lib/change-notes/2022-04-08-flow-state-barriers.md:4]:
```suggestion
* The recently added flow-state versions of `isBarrierIn`, `isBarrierOut`, `isSanitizerIn`, and `isSanitizerOut` in the data flow and taint tracking libraries have been removed.
```

File: java/ql/lib/change-notes/2022-04-08-flow-state-barriers.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+---
+category: breaking
+---
+The recently added flow-state versions of `isBarrierIn`, `isBarrierOut`, `isSanitizerIn`, and `isSanitizerOut` in the data flow and taint tracking libraries have been removed.
```
Comment (jketema) [inline java/ql/lib/change-notes/2022-04-08-flow-state-barriers.md:4]:
```suggestion
* The recently added flow-state versions of `isBarrierIn`, `isBarrierOut`, `isSanitizerIn`, and `isSanitizerOut` in the data flow and taint tracking libraries have been removed.
```

File: python/ql/lib/change-notes/2022-04-08-flow-state-barriers.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+---
+category: breaking
+---
+The recently added flow-state versions of `isBarrierIn`, `isBarrierOut`, `isSanitizerIn`, and `isSanitizerOut` in the data flow and taint tracking libraries have been removed.
```
Comment (jketema) [inline python/ql/lib/change-notes/2022-04-08-flow-state-barriers.md:4]:
```suggestion
* The recently added flow-state versions of `isBarrierIn`, `isBarrierOut`, `isSanitizerIn`, and `isSanitizerOut` in the data flow and taint tracking libraries have been removed.
```

File: ruby/ql/lib/change-notes/2022-04-08-flow-state-barriers.md [hunk 6]
```diff
@@ -0,0 +1,4 @@
+---
+category: breaking
+---
+The recently added flow-state versions of `isBarrierIn`, `isBarrierOut`, `isSanitizerIn`, and `isSanitizerOut` in the data flow and taint tracking libraries have been removed.
```
Comment (jketema) [inline ruby/ql/lib/change-notes/2022-04-08-flow-state-barriers.md:4]:
```suggestion
* The recently added flow-state versions of `isBarrierIn`, `isBarrierOut`, `isSanitizerIn`, and `isSanitizerOut` in the data flow and taint tracking libraries have been removed.
```

**Review by jketema — CHANGES_REQUESTED** [review]
Per change note documentation

**Review by jketema — APPROVED** [review]

**Comment by geoffw0** [comment]
What's the story here?  I'm using `isBarrierOut(DataFlow::Node node, string flowstate)` in an upcoming PR.

**Comment by MathiasVP** [comment]
> What's the story here? I'm using `isBarrierOut(DataFlow::Node node, string flowstate)` in an upcoming PR.

It turns out that this is too difficult to implement correctly. See the conversation linked in the PR description: https://github.com/github/codeql/pull/8435#issuecomment-1087530306

**Comment by geoffw0** [comment]
I think I can get by using `isBarrier(Node node, FlowState state)`, with probably little or no loss of accuracy.  Is that the right approach?

**Comment by MathiasVP** [comment]
> I think I can get by using `isBarrier(Node node, FlowState state)`, with probably little or no loss of accuracy. Is that the right approach?

`isBarrier(node, state)` isn't going anywhere, no. I can't tell whether it's the right approach (since I don't know what your use-case is!), but it's definitely an API that's supported by the dataflow library.

**Comment by jketema** [comment]
@aschackmull https://github.com/github/codeql/pull/8702 has been merged. I'm not sure if you want to rebase this to get all the tests green or not, so I leave merging of this PR to you.

**Comment by aschackmull** [comment]
> @aschackmull #8702 has been merged. I'm not sure if you want to rebase this to get all the tests green or not, so I leave merging of this PR to you.

Thanks, I've rebased the PR.

**Review by jketema — APPROVED** [review]

**Pull request merged**